src/lib.rs

```diff
@@ -126,15 +126,49 @@ pub trait Renderable: Send + Sync {
     fn render(&self, context: &mut Context) -> Result<Option<String>>;
 }
 
+pub type TemplateLocation = Path;
+pub trait FileRepository {
+    fn read_template_file(&self, path: &TemplateLocation) -> Result<String>;
```

Comment:
You bring up some good points.  This is going to be a bigger decision that we should probably plan out in a separate RFC issue.  How about we keep to just returning a string for now?